### PR TITLE
En_Dy_Extra OK and matching

### PR DESCRIPTION
**Pull request by fullgrowngaming**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
 #include "z_en_dy_extra.h"
+#include "vt.h"
 
 #define FLAGS 0x00000030
 
 #define THIS ((EnDyExtra*)thisx)
 
+extern UNK_TYPE D_0601BFB0;
+extern Gfx* D_0601C160;
+
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:16]: not a huge deal but i like to keep the externs underneath all the data and stuff

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 2]
```diff
@@ -21,15 +33,102 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+void EnDyExtra_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnDyExtra_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDyExtra* this = THIS;
+
+    osSyncPrintf("\n\n");
+    //"Big fairy effect"
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 大妖精効果 ☆☆☆☆☆ %d\n" VT_RST, this->actor.params);
+    this->unk_150 = this->actor.params;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:46]: you can name unk_150 params if you wanted

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 3]
```diff
@@ -21,15 +33,102 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+    this->actionFunc = func_809FF7AC;
+}
+
+void func_809FF7AC(EnDyExtra* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxF(&this->actor.gravity, 0.0f, 0.1f, 0.005f);
+    if (this->actor.posRot.pos.y < -55.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+    if (this->unk_154 == 0) {
+        if (this->unk_152 != 0) {
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:63]: these two can be anded together

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 4]
```diff
@@ -21,15 +33,102 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+            this->unk_154 = 0xC8;
+            this->actionFunc = func_809FF840;
+        }
+    }
+}
+
+void func_809FF840(EnDyExtra* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxF(&this->actor.gravity, 0.0f, 0.1f, 0.005f);
+    if (this->unk_154 == 0 || this->unk_158 < 0.02f) {
+        Actor_Kill(&this->actor);
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:73]: with actor kill usually an early return makes more sense vs and else cause its like the condition to stop actor execution entirely, so everything just ends there
both are completely valid tho but just to keep the pattern consistent we can change it here

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 5]
```diff
@@ -21,15 +33,102 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+    this->actor.scale.y = this->unk_15C.y;
+    this->actor.scale.z = this->unk_15C.z;
+    Audio_PlayActorSound2(&this->actor, NA_SE_PL_SPIRAL_HEAL_BEAM - SFX_FLAG);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(&this->actor);
+}
+
+void EnDyExtra_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Color_RGBA8_n D_809FFC40[] = { { 255, 255, 170, 255 }, { 255, 255, 170, 255 } };
+    static Color_RGBA8_n D_809FFC48[] = { { 255, 100, 255, 255 }, { 100, 255, 255, 255 } };
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:96]: you can name these primColors and envColors or something along those lines

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 6]
```diff
@@ -21,15 +33,102 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+    static u8 D_809FFC50[] = { 0x02, 0x01, 0x01, 0x02, 0x00, 0x00, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x02,
+                               0x01, 0x00, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x01, 0x02 };
+    EnDyExtra* this = THIS;
+    s32 pad;
+    GraphicsContext* localGfxCtx = globalCtx->state.gfxCtx;
+    Vtx* data = (Vtx*)SEGMENTED_TO_VIRTUAL(&D_0601BFB0);
+    s32 i;
+    u8 unk[3];
+    GraphicsContext* gfxCtx = localGfxCtx;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:105]: for the sake of the macro stuff i would just declare the gfxCtx temp and then assign it later like you already have there

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 7]
```diff
@@ -21,15 +30,103 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+        if (D_809FFC50[i]) {
+            data[i].v.cn[3] = unk[D_809FFC50[i]];
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Update.s")
+    gfxCtx = localGfxCtx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Draw.s")
+    Graph_OpenDisps(dispRefs, localGfxCtx, "../z_en_dy_extra.c", 0x126);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:118]:
```suggestion
    Graph_OpenDisps(dispRefs, localGfxCtx, "../z_en_dy_extra.c", 294);
```

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 8]
```diff
@@ -21,15 +30,103 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+extern Gfx* D_0601C160;
+
+void EnDyExtra_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnDyExtra_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDyExtra* this = THIS;
+
+    osSyncPrintf("\n\n");
+    //"Big fairy effect"
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:44]: Make sure to run the formatter on the file
```suggestion
    // "Big fairy effect"
```

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 9]
```diff
@@ -21,15 +30,103 @@ const ActorInit En_Dy_Extra_InitVars = {
     (ActorFunc)EnDyExtra_Update,
     (ActorFunc)EnDyExtra_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Init.s")
+extern UNK_TYPE D_0601BFB0;
+extern Gfx* D_0601C160;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:35]:
```suggestion
extern Gfx D_0601C160[];
```

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 10]
```diff
@@ -21,15 +30,103 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, globalCtx->state.frames * 2, 0, 0x20, 0x40, 1,
+                                globalCtx->state.frames, globalCtx->state.frames * -8, 0x10, 0x10));
+    gDPPipeSync(gfxCtx->polyXlu.p++);
+    gSPMatrix(gfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_dy_extra.c", 307),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gDPSetPrimColor(gfxCtx->polyXlu.p++, 0, 0x80, primColors[this->params].r, primColors[this->params].g,
+                    primColors[this->params].b, 255);
+    gDPSetEnvColor(gfxCtx->polyXlu.p++, envColors[this->params].r, envColors[this->params].g, envColors[this->params].b,
+                   128);
+    gSPDisplayList(gfxCtx->polyXlu.p++, &D_0601C160);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:130]:
```suggestion
    gSPDisplayList(gfxCtx->polyXlu.p++, D_0601C160);
```

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.h [hunk 11]
```diff
@@ -6,9 +6,17 @@
 
 struct EnDyExtra;
 
+typedef void (*EnDyExtraActionFunc)(struct EnDyExtra*, GlobalContext*);
+
 typedef struct EnDyExtra {
     /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x28];
+    /* 0x014C */ EnDyExtraActionFunc actionFunc;
+    /* 0x0150 */ s16 params;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.h:14]:
```suggestion
    /* 0x0150 */ s16 type;
```

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 12]
```diff
@@ -21,15 +30,103 @@ const ActorInit En_Dy_Extra_InitVars = {
     (ActorFunc)EnDyExtra_Update,
     (ActorFunc)EnDyExtra_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Init.s")
+extern UNK_TYPE D_0601BFB0;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:34]:
```suggestion
extern Vtx D_0601BFB0[];
```

File: src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c [hunk 13]
```diff
@@ -21,15 +30,103 @@ const ActorInit En_Dy_Extra_InitVars = {
...
+
+void EnDyExtra_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Color_RGBA8_n primColors[] = { { 255, 255, 170, 255 }, { 255, 255, 170, 255 } };
+    static Color_RGBA8_n envColors[] = { { 255, 100, 255, 255 }, { 100, 255, 255, 255 } };
+    static u8 D_809FFC50[] = { 0x02, 0x01, 0x01, 0x02, 0x00, 0x00, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x02,
+                               0x01, 0x00, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x01, 0x02 };
+    EnDyExtra* this = THIS;
+    s32 pad;
+    GraphicsContext* localGfxCtx = globalCtx->state.gfxCtx;
+    Vtx* data = (Vtx*)SEGMENTED_TO_VIRTUAL(&D_0601BFB0);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c:100]:
```suggestion
    Vtx* vertices = SEGMENTED_TO_VIRTUAL(D_0601BFB0);
```

**Review by Roman971 — APPROVED** [review]

**Pull request merged**